gadgets/src/mul_add.rs

```diff
@@ -0,0 +1,438 @@
+//! Chip that implements instructions to check: a * b + c == d (mod 2^256) where
```

Comment:
The difference is that the `math_gadget.rs` in `evm_circuit` crate builds constraints using the [`ConstraintBuilder`](https://github.com/scroll-tech/zkevm-circuits/blob/feat/exp/zkevm-circuits/src/evm_circuit/util/constraint_builder.rs#L241) type which is specifically designed for use within the EVM Circuit.

The [`gadgets`](https://github.com/scroll-tech/zkevm-circuits/tree/feat/exp/gadgets) crate on the other hand uses the [`ConstraintSystem`](https://github.com/scroll-tech/zkevm-circuits/blob/feat/exp/gadgets/src/mul_add.rs#L59) provided by `halo2-proofs` and hence can be used in a circuit like Copy Circuit or Exponentiation Circuit.

The refactoring of `gadgets` crate and maintaining them along with the EVM Circuit's util gadgets was proposed in [this issue](https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/444) because even the MPT and Keccak Circuits would've required gadgets such as `IsZero` or `LessThan`, etc.